### ISSUE TITLE
Ground truth environment map?

### ISSUE THREAD

**Issue by hengyelyu** [TMT22/HumanOLAT#1]
Thank you for your contribution in providing this dataset! 

In the processed dataset, each instance contains 364 frames per camera under different lighting conditions, which is different from the total number of lighting conditions.

 I would like to know where the corresponding ground-truth environment lighting map for these images can be found.

Thank you!

**Comment by TMT22** [comment]
OLAT environment maps can be found under `./code/olat_relight/OLAT_EnvMaps`. They are ordered the same as the OLAT frames (beginning with the first fullbright in the sequence, if I remember correctly, frame 15?) with every black envmap representing a fullbright frame. Other than that, `read_OLAT_info(...)` in `./code/utils/metadata_readers.py` also gives you a point light position for each frame.

The maps for the frames captures under the enviroment light should be unter `./code/olat_relight/example_envmaps`. They are also in the same order (sorted alphabetically) as the frames in the dataset, e.g. bedroom.exr should correspond to the first envmap frame. However, since it is difficult to match the result of relighting with OLAT images with frames captured under direct envmaps (e.g. due to differences in light intensities), I recommend to just generate as many frames as needed using envmaps of your choice using the OLATs (or, ideally train directly on the OLATs).

For more detailed information, please view the datasheet and the provided example code. : )

**Comment by fattypiggy** [comment]
Thanks for your answer @TMT22 ,
I also noticed that the last two environment maps in `code/olat_relight/OLAT_EnvMaps` are completely black. I know a completely black environment map refers to full light (or could it also be understood as flat light?), but when I look at the last two OLAT images in any directory, they are noticeably darker than the other full-light images. What could be causing this?

**Comment by fattypiggy** [comment]
Here is my understanding of the structure of HumanOLAT Dataset 
- 21 subjects 
- 3 poses * 20 + 4 poses * 1(SUBJECT_C058) 
- 40 cameras 
- 364 images per pose 
  - Flat light * 1 
  - Gradient color * 2 
  - Env map light * 11 
  - OLAT * 350 
    - Light * 331 
    - Flat light * 17 
    - ?? * 2

**Comment by TMT22** [comment]
As far as I can see, that structure seems to be correct. To my knowledge, those last two frames are also just flat-lits similar to the fullbright frames, but with a lower light intensity (more precisely, the default lighting in our stage after the sequence ran through). For using the dataset, those two frames can be ignored (the provided dataloaders should also just ignore those files at the end).

Effectively, those two frames are artifacts of the processing pipeline. The raw streams coming from cameras are not immediately synchronized (e.g. some cameras recorded 10 frames less than others, for some the intended sequence starts only after 2 frames, etc.). The script we use for synchronization finds the correct frame where the sequence begins, then grabs the next 364 frames (which includes those two low light frames to check consistency) and cuts the rest. Those two end frames however are never removed at a later stage and essentially just processed as all the other frames.

**Comment by fattypiggy** [comment]
Thanks for your reply.
I also found two issues related to the `processed_data`:
1. SUBJECT_C075 POSE_01 only contains 361 images.
2. SUBJECT_C028 folder contains 728 images which is double the size of 364, and I found there is another subject in the folder, but the results are incorrect, masks mismatch.

And one more question, do you treat **full light** images as **flat lit** images in relighting task? Or do I need to get the **albedo** by my own using other models?

**Comment by TMT22** [comment]
Ah, thanks for the info. : )
Regarding SUBJECT_C028, the additional subject there is C032 but with masks from C028. Likely, something went wrong when I was generating the processed files and the error went unnoticed because the files coexist without interfering with each other. Here, you can ignore any files labeled C032*
For SUBJECT_C075 POSE_01, yeah, a single OLAT frame at the end is missing. Need to partly reprocess the sequence to fix that, likely the extraction stopped prematurely (worst case, cameras did not record that single frame). 

For your final question, those images are flat lit, but to my knowledge technically not perfect albedo/basecolor, since e.g. specular light is not filtered out here. However, certain works such as RNHA [1] show that you can use it as an approximation to albedo in a relighting framework.   

[1] https://openaccess.thecvf.com/content/CVPR2023/papers/Zhou_Relightable_Neural_Human_Assets_From_Multi-View_Gradient_Illuminations_CVPR_2023_paper.pdf

**Comment by fattypiggy** [comment]
Hi @TMT22 , 
Can I assume you and [FaceOLAT](https://github.com/prraoo/FaceOLAT) used the same light stage to capture the image? That is, all configurations are identical, including camera position, camera intrinsic parameter settings, and the brightness of the 331 lights. Could you please provide me with the light brightness information? I want to relit the image. Thank you.
